docs(toh-5/dart): review edits and code cleanup

chalin opened the pull request:
- Adjust text which used to say "create a new 'styles.css'" when we've been carrying it around since QuickStart.
- The HTTP tutorial chapter is no longer "forthcoming".
- This commit also adds a missing docregion to a toh-4 file that is referrenced in toh-5.
- Fixed prose over indentation (which appeared in over half the file).
- Copyedits.

Code:
- Add missing return types.
- Handle invalid numeric route hero ids.
- Rename HERO mockHeroes as used in earlier tutorial chapters.
- Use Dartisms (cleanup from ts2dart transpilation).

Contributing to #1628.

chalin commented:
@kwalrath @thso : prose and code changes are ready for review.

kwalrath commented:
Where is styles_1.css referred to? I remember this being a weird situation, but I don't remember the details.

kwalrath commented on public/docs/dart/latest/tutorial/toh-pt5.jade:
```diff
@@ -48,648 +48,658 @@ p Run the #[+liveExampleLink2('', 'toh-5')] for this part.
-    * add routing
-    * create a new `DashboardComponent`
-    * tie the *Dashboard* into the navigation structure.
+  * turn `AppComponent` into an application shell that only handles navigation,
```
If we're editing this, I'd use sentence caps but **not** add commas at the end. (Periods would be fine, though.) I'd also add a colon after "plan". For example:

Here's our plan:

* Turn ... navigation.
* ...
* Tie ... structure.

chalin replied:
That is my preference too. Done.

kwalrath commented on public/docs/dart/latest/tutorial/toh-pt5.jade:
```diff
@@ -48,648 +48,658 @@ p Run the #[+liveExampleLink2('', 'toh-5')] for this part.
-    * rename `app_component.dart` file to `heroes_component.dart`.
-    * rename the `AppComponent` class to `HeroesComponent`.
-    * rename the selector `my-app` to `my-heroes`.
+  The steps are: rename the
```
Move the ":" to the end of the line.

chalin replied:
Fixed.

kwalrath commented on public/docs/dart/latest/tutorial/toh-pt5.jade:
```diff
@@ -48,648 +48,658 @@ p Run the #[+liveExampleLink2('', 'toh-5')] for this part.
-    * rename the `AppComponent` class to `HeroesComponent`.
-    * rename the selector `my-app` to `my-heroes`.
+  The steps are: rename the
+  * `app_component.dart` file to `heroes_component.dart`,
```
Remove the punctuation at the end of these lines.

chalin replied:
Fixed.

kwalrath commented on public/docs/dart/latest/tutorial/toh-pt5.jade:
```diff
@@ -48,648 +48,658 @@ p Run the #[+liveExampleLink2('', 'toh-5')] for this part.
+  1. *Cut-and-paste* the styles contents into a new `heroes_component.css` file.
+  1. *Set* the component metadata's `templateUrl` and `styleUrls` properties to refer to both files.
+  
+  We no longer display the `HeroDetailComponent` in the `HeroesComponent` template because we're navigating to it.
```
Let's reword this a bit to clarify what "it" is. Also, maybe you could show the imports, since you mention them? Or don't show them, but don't mention them. (If you do mention them, try to reword "might as well".) Maybe:

Because the template for `HeroesComponent` no longer _uses_ `HeroDetailComponent` directly &mdash;
instead using the router to _navigate_ to it &mdash; we can remove `HeroDetailComponent` from the `directives` list. That list is now empty, so we can remove the `directives` argument. The revised `@Component` looks like this:

chalin replied:
I like your suggestion. Fixed.

kwalrath commented on public/docs/dart/latest/tutorial/toh-pt5.jade:
```diff
@@ -48,648 +48,658 @@ p Run the #[+liveExampleLink2('', 'toh-5')] for this part.
+  Add a new file named `styles.css` in the root folder, if there isn't one already.
+  Ensure that it contains the [master styles given here](!{styles_css}).
+
+  Also ensure this stylesheet is referenced in the traditional manner within `index.html`.
```
"is referenced" is passive voice. Maybe:

Also, edit `index.html` to refer to this stylesheet:

chalin replied:
Fixed.

kwalrath commented on public/docs/dart/latest/tutorial/toh-pt5.jade:
```diff
@@ -48,648 +48,658 @@ p Run the #[+liveExampleLink2('', 'toh-5')] for this part.
-    Verify that we have the following structure:
+  ## Application structure and code
+p.
+  Review the sample source code in the #[+liveExampleLink2('', 'toh-5')] for this part.
```
Why change chapter to part? I thought they were consistently using "chapter" for each page?

chalin replied:
It seems that in the tutorial chapters are referred to as parts. Consider the following excerpt from ... [Chapter/Part 2](https://angular.io/docs/dart/latest/tutorial/toh-pt2.html):

> Before we continue with **Part** 2 of the Tour of Heroes, let’s verify we have the following structure after **Part** 1. If not, we’ll need to go back to **Part** 1 and figure out what we missed.

But I agree that in this case "chapter" might make more sense. Changing it back to "chapter".

kwalrath commented:
The changes look good, with the exception of what I commented on. (This page needs a good copy edit, but I tried to avoid looking at anything you didn't change!)

chalin commented:
I agree about further copy edits. For now, I have addressed the points that you raised, and made the changes to both the TS and Dart sides so that they match.

kwalrath commented:
LGTM. Merging now...
